src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller.Session_FilterModel.cs

```diff
@@ -457,7 +457,7 @@ public FilterResult(CompletionItem completionItem, string filterText, bool match
                 if (filterReason == CompletionFilterReason.BackspaceOrDelete &&
                     trigger.Kind == CompletionTriggerKind.Deletion)
                 {
-                    return item.FilterText.GetCaseInsensitivePrefixLength(filterText) > 0;
+                    return item.FilterText.GetCaseInsensitivePrefixLength(filterText) >= Math.Min(item.FilterText.Length, filterText.Length);
```

Comment:
My bad--this should just be an equality check.

When you type backspace, the idea is that completion should try to match items where the filter text is a prefix of the item text. In order to actually be a prefix match, all of the filter text needs to be a prefix. This code used to require that only 1 character of the filter text appear in the item text which would cause us to generate bad matches. We actually need the "case insensitive prefix match" length to be as long as whichever of those strings was shorter. 

This fixes https://github.com/dotnet/roslyn/issues/14704